mlflow/store/job/abstract_store.py

```diff
@@ -0,0 +1,137 @@
+from abc import ABCMeta, abstractmethod
+from typing import Iterator
+
+from mlflow.entities._job import Job
+from mlflow.entities._job_status import JobStatus
+from mlflow.utils.annotations import developer_stable
+
+
+@developer_stable
+class AbstractJobStore:
+    """
+    Abstract class that defines API interfaces for storing Job metadata.
+    """
+
+    __metaclass__ = ABCMeta
+
+    def __init__(self):
+        """
+        Empty constructor. This is deliberately not marked as abstract, else every derived class
+        would be forced to create one.
+        """
+
+    @abstractmethod
+    def create_job(self, function_fullname: str, params: str, timeout: int | None = None) -> Job:
+        """
+        Create a new job with the specified function and parameters.
+
+        Args:
+            function_fullname: The full name of the function to execute
+            params: The job parameters as a string
+            timeout: The job execution timeout in seconds
+
+        Returns:
+            Job entity instance
+        """
+
+    @abstractmethod
+    def start_job(self, job_id: str) -> None:
+        """
+        Start a job by setting its status to RUNNING.
+
+        Args:
+            job_id: The ID of the job to start
+        """
+
+    @abstractmethod
+    def reset_job(self, job_id: str) -> None:
+        """
+        Reset a job by setting its status to PENDING.
+
+        Args:
+            job_id: The ID of the job to re-enqueue.
+        """
+
+    @abstractmethod
+    def finish_job(self, job_id: str, result: str) -> None:
+        """
+        Finish a job by setting its status to DONE and setting the result.
+
+        Args:
+            job_id: The ID of the job to finish
+            result: The job result as a string
+        """
+
+    @abstractmethod
+    def set_job_timeout(self, job_id: str) -> None:
+        """
+        Set a job status to Timeout.
```

Comment:
this does sound like a job has a timeout parameter and this method updates it.

Reply:
agree. renamed it to `mark_job_timed_out`